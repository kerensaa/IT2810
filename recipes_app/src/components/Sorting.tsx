import { FormControl, InputLabel, MenuItem, Select, SelectChangeEvent } from '@mui/material';
import '../styling/Sorting.css';

interface SortingProps {
  sortingOption: string;
  onSortChange: (value: string) => void;
}

function Sorting(props: SortingProps) {
  const handleSort = (event: SelectChangeEvent) => {
    props.onSortChange(event.target.value);
    const newUrl = new URL(window.location.href);
    newUrl.searchParams.set('sort', event.target.value);
    window.history.pushState({}, '', newUrl.toString());
    window.location.reload();
  };

  return (
    <div className="sort_select">
      <FormControl>
        <InputLabel>Sort</InputLabel>
        <Select
          labelId="demo-simple-select-label"
          id="demo-simple-select"
          label="Age"
          value={props.sortingOption}
          onChange={handleSort}
        >
          <MenuItem key={'name'} value={'name'}>
            Name
          </MenuItem>
          <MenuItem key={'prep_time'} value={'prep_time'}>
            Prep Time
          </MenuItem>
          <MenuItem key={'default'} value={'default'}>
            No Sorting
          </MenuItem>
        </Select>
      </FormControl>
    </div>
  );
}

export default Sorting;
