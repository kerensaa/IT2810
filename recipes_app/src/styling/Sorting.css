.sort_select {
  margin-right: 10px;
}
