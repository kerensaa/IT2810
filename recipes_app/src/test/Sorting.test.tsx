import { fireEvent, render, screen } from '@testing-library/react';
import { LinkProps } from 'react-router-dom';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import Sorting from '../components/Sorting';

vi.mock('react-router-dom', () => ({
  Link: ({ to, children, ...props }: LinkProps) => (
    <a href={typeof to === 'string' ? to : undefined} {...props}>
      {children}
    </a>
  ),
}));

describe('Renders Sorting correctly', async () => {
  beforeEach(() => {
    render(<Sorting sortingOption="default" onSortChange={() => {}} />);
  });
  it('Dropdown sort component works correctly', () => {
    const nosort = screen.getByText('No Sorting');
    expect(nosort).not.toBeNull;
    fireEvent.mouseDown(nosort);
    fireEvent.click(screen.getByText('Prep Time'));
    const preptime = screen.getByText('Prep Time');
    expect(preptime).not.toBeNull;
    fireEvent.mouseDown(preptime);
    fireEvent.click(screen.getByText('Name'));
    const name = screen.getByText('Name');
    expect(name).not.toBeNull;
  });
});

describe('Snapshot testing of Sorting', async () => {
  it('should perform snapshot test Sorting', async () => {
    const navbar = render(<Sorting sortingOption="default" onSortChange={() => {}} />);
    expect(navbar).toMatchSnapshot();
  });
});
