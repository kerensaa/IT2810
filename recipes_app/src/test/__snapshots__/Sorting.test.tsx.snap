// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot testing of Sorting > should perform snapshot test Sorting 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <div
        class="sort_select"
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
          >
            Sort
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-label demo-simple-select"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="demo-simple-select"
              role="button"
              tabindex="0"
            >
              No Sorting
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="default"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Sort
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiMenu-root MuiPopover-root MuiModal-root css-10nakn3-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="menu-"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; webkit-transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 195ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiMenu-paper MuiPopover-paper MuiMenu-paper css-3dzjca-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
        style="opacity: 0; transform: scale(0.75, 0.5625); min-width: 0; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
        tabindex="-1"
      >
        <ul
          aria-labelledby="demo-simple-select-label"
          class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
          role="listbox"
          tabindex="-1"
        >
          <li
            aria-selected="false"
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            data-value="name"
            role="option"
            tabindex="-1"
          >
            Name
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            aria-selected="false"
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            data-value="prep_time"
            role="option"
            tabindex="-1"
          >
            Prep Time
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            >
              <span
                class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
                style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
              >
                <span
                  class="MuiTouchRipple-child"
                />
              </span>
            </span>
          </li>
          <li
            aria-selected="true"
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters Mui-selected MuiMenuItem-root MuiMenuItem-gutters Mui-selected css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            data-value="default"
            role="option"
            tabindex="0"
          >
            No Sorting
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
    <div>
      <div
        class="sort_select"
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
          >
            Sort
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-label demo-simple-select"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="demo-simple-select"
              role="button"
              tabindex="0"
            >
              No Sorting
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="default"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Sort
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sort_select"
    >
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
        >
          Sort
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-label demo-simple-select"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="demo-simple-select"
            role="button"
            tabindex="0"
          >
            No Sorting
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="default"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Sort
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
